dex/networks/zec/tx.go

```diff
@@ -471,19 +725,33 @@ func writeOutPoint(w io.Writer, version int32, op *wire.OutPoint) error {
 	return putUint32(w, op.Index)
 }
 
-// putUint32 writes a little-endian encoded uint32 to the Writer.
-func putUint32(w io.Writer, v uint32) error {
+func uint32Bytes(v uint32) []byte {
 	b := make([]byte, 4)
 	binary.LittleEndian.PutUint32(b, v)
-	_, err := w.Write(b)
+	return b
+}
+
+// putUint32 writes a little-endian encoded uint32 to the Writer.
+func putUint32(w io.Writer, v uint32) error {
+	_, err := w.Write(uint32Bytes(v))
 	return err
 }
 
-// putUint32 writes a little-endian encoded uint64 to the Writer.
-func putUint64(w io.Writer, v uint64) error {
+func uint64Bytes(v uint64) []byte {
 	b := make([]byte, 8)
 	binary.LittleEndian.PutUint64(b, v)
-	_, err := w.Write(b)
+	return b
+}
+
+func int64Bytes(v int64) []byte {
+	buf := new(bytes.Buffer)
+	binary.Write(buf, binary.LittleEndian, v)
```

Comment:
This just does `order.PutUint16(bs, uint16(v))` internally, with it's own slice alloc.  I'd do it directly to skip the extra alloc, the err return, and overhead of `Write`.